ERCS/erc-7715.md

```diff
@@ -0,0 +1,601 @@
+
+`chainId` is field that identifies that chain with EIP-155 which these permissions are contextual and all addressed can be found for the accounts and permissions.
+
+`signer` is a field that identifies the key or account associated with the session or alternatively the wallet will manage the session. See the “Wallet-managed Sessions” section for details.
+
+`permissions` is a list of permissions with determined policies. See “Policy Types” section for details.
+
+Each permission can be either `required` or not. If a permission is `required`, the wallet MUST grant the permission if it approves the request. Otherwise, the wallet may reject the permission even if it approves the request (presumably because it grants other permissions).
+
+`expiry` is a UNIX timestamp (in seconds) that specifies the time by which this session MUST expire.
+
+Request example:
+
+```tsx
+{
+  account: '0x...'
+
+  chainId: 123,
+
+  signer: {
+    type: 'key',
+    data: {
+      id:'did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme',
+    }
+  },
+
+  permissions: [
+    {
+      type: 'native-token-transfer',
+      policies: [
+        {
+          type: 'gas-limit',
+          data: {
+            limit: '0x...',
+          },
+        },
+        {
+          type: 'call-limit',
+          data: {
+            count: 1,
+          },
+        },
+      ],
+      required: true,
+      data: {
+        address: '0x...',
+        allowance: '500000000'
+      }
+    },
+    {
+      type: 'erc20-token-transfer',
+      policies: [
+        {
+          type: 'gas-limit',
+          data: {
+            limit: '0x...',
+          },
+        }
+      ],
+      required: false,
+      data: {
+        address: '0x...',
+        allowance: '2000000000'
+      }
+    }
+  ],
+
+  expiry: 1577840461
+}
+```
+
+#### Response Specification
+
+```tsx
+type GrantPermissionsResponse = {
+  grantedPermissions: Permission[];
+
+  expiry: number;
+
+  signerData: {
+    userOpBuilder?: `0x${string}`;
+    managerAddress?: `0x${string}`;
+  };
+
+  factory?: `0x${string}`;
+
+  factoryData?: `0x${string}`;
+
+  permissionsContext: string;
+};
+```
+
+- `grantedPermissions` is a list of permissions that were granted. The wallet MUST grant `required` permissions if the request was approved.
+- `expiry` is the expiry time that was granted. The wallet MAY grant a different expiry time than the one that the DApp requested.
+- `factory` and `factoryData` are as defined in [ERC-4337](./eip-4337.md). They are either both specified, or none. If the account has not yet been deployed, the wallet MUST return `factory` and `factoryData`, and the DApp MUST deploy the account by calling the `factory` contract with `factoryData` as the calldata.
+- `permissionsContext` is an opaque string that identifies the session. It MAY be the `context` as defined in [ERC-7679](./eip-7679.md) and [ERC-7710](./eip-7710.md). See “Rationale” for details.
```

Comment:
Missing context on the `userOpBuilder` and `managerAddress` parameters:
```suggestion
- `grantedPermissions` is a list of permissions that were granted. The wallet MUST grant `required` permissions if the request was approved.
- `expiry` is the expiry time that was granted. The wallet MAY grant a different expiry time than the one that the DApp requested.
- `signerData.userOpBuilder` is as defined in [ERC-7679](./eip-7679.md).
- `signerData.managerAddress` is as defined in [ERC-7710](./eip-7710.md).
- `factory` and `factoryData` are as defined in [ERC-4337](./eip-4337.md). They are either both specified, or none. If the account has not yet been deployed, the wallet MUST return `factory` and `factoryData`, and the DApp MUST deploy the account by calling the `factory` contract with `factoryData` as the calldata.
- `permissionsContext` is an opaque string that identifies the session. It MAY be the `context` as defined in [ERC-7679](./eip-7679.md) and [ERC-7710](./eip-7710.md). See “Rationale” for details.
```